src/Component/DataRetentionSettings.elm

```diff
@@ -0,0 +1,332 @@
+            , focusIfChecked "errors-inp" isChecked
+            )
+
+        IdleDocTimeoutErrorInputChanged input ->
+            ( { model | idleDocTimeoutError = updateCurrentValue input model.idleDocTimeoutError }, Cmd.none )
+
+        ImmediateTrashChecked isChecked ->
+            ( { model | immediateTrash = isChecked }, Cmd.none )
+
+        _ ->
+            ( model, Cmd.none )
+
+
+
+-- Validation --
+
+
+isFieldValid : DataRetentionField -> Bool
+isFieldValid field =
+    if field.isChecked then
+        String.toInt field.currentValue
+            |> Maybe.isJust
+
+    else
+        True
+
+
+isFormValid : Model -> Bool
+isFormValid model =
+    List.all isFieldValid
+        [ model.idleDocTimeoutPreparation
+        , model.idleDocTimeoutClosed
+        , model.idleDocTimeoutCanceled
+        , model.idleDocTimeoutTimedout
+        , model.idleDocTimeoutRejected
+        , model.idleDocTimeoutError
+        ]
+
+
+isFieldChanged : DataRetentionField -> Bool
+isFieldChanged field =
+    if field.isChecked then
+        String.toInt field.currentValue /= field.initialValue
+
+    else
+        Maybe.isJust field.initialValue
+
+
+isFormChanged : Model -> Bool
+isFormChanged model =
+    List.any isFieldChanged
+        [ model.idleDocTimeoutPreparation
+        , model.idleDocTimeoutClosed
+        , model.idleDocTimeoutCanceled
+        , model.idleDocTimeoutTimedout
+        , model.idleDocTimeoutRejected
+        , model.idleDocTimeoutError
+        ]
+        || (model.initialImmediateTrash /= model.immediateTrash)
+
+
+
+-- View --
+
+
+checkBoxField : String -> Bool -> (Bool -> Msg) -> (String -> Msg) -> String -> DataRetentionField -> Html Msg
+checkBoxField id_ isInherited toCheckedMsg toInputMsg labelText field =
+    div [ class "flex items-center mb-4" ]
+        [ input
+            [ checked field.isChecked
+            , Events.onCheck <| toCheckedMsg
+            , disabled isInherited
+            , id <| id_ ++ "-cbx"
+            , type_ "checkbox"
+            , value ""
+            , class "w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 focus:ring-2"
+            ]
+            []
+        , span [ class "ms-2 text-sm font-medium" ]
+            [ text labelText
+            , input
+                [ type_ "number"
+                , id <| id_ ++ "-inp"
+                , value field.currentValue
+                , disabled <| not field.isChecked || isInherited
+                , Events.onInput <| toInputMsg
+                , Attrs.min "1"
+                , classList [ ( "border-red-500", Maybe.isNothing <| String.toInt field.currentValue ) ]
+                , class "disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 disabled:shadow-none w-20 appearance-none border rounded py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
+                ]
```

Comment:
At least on Chrome, I was able to make it accept "0" and "-1" just by typing it in the field with they keyboard.
How would you prevent this?